### DISCUSSION TITLE
Update Not Working

### DISCUSSION THREAD

**Discussion by Rochambeaux** [mcguirepr89/BirdNET-Pi#940] [Q&A]
Please see log below.  is there a way to command line updates to get around this?

fatal: pathspec 'privacy_server.py' did not match any files
Saved working directory and index state WIP on main: df12e6b Checkout known-working commit
hint: Pulling without specifying how to reconcile divergent branches is
hint: discouraged. You can squelch this message by running one of the following
hint: commands sometime before your next pull:
hint: 
hint:   git config pull.rebase false  # merge (the default strategy)
hint:   git config pull.rebase true   # rebase
hint:   git config pull.ff only       # fast-forward only
hint: 
hint: You can replace "git config" with "git config --global" to set a default
hint: preference for all repositories. You can also pass --rebase, --no-rebase,
hint: or --ff-only on the command line to override the configured default per
hint: invocation.
Committer identity unknown

*** Please tell me who you are.

Run

  git config --global user.email "you@example.com"
  git config --global user.name "Your Name"

to set your account's default identity.
Omit --global to set the identity only in this repository.

fatal: unable to auto-detect email address (got 'jbr@birdnetpi.(none)')
On branch main
Your branch and 'origin/main' have diverged,
and have 5 and 1055 different commits each, respectively.
  (use "git pull" to merge the remote branch into yours)

Changes not staged for commit:
  (use "git add ..." to update what will be committed)
  (use "git restore ..." to discard changes in working directory)
	modified:   analyze.py
	modified:   daily_plot.py
	modified:   plotly_streamlit.py
	modified:   server.py

Untracked files:
  (use "git add ..." to include in what will be committed)
	../BirdDB.txt
	../HUMAN.txt
	../analyzing_now.txt
	../apprise.txt
	../birdnet/
	../exclude_species_list.txt
	../firstrun.ini
	../include_species_list.txt
	birds.db
	disk_check_exclude.txt
	exclude_species_list.txt
	include_species_list.txt
	labels.txt
	lastrun.txt
	thisrun.txt
	../templates/avahi-alias@.service
	../templates/birdnet_analysis.service
	../templates/birdnet_log.service
	../templates/birdnet_recording.service
	../templates/birdnet_server.service
	../templates/birdnet_stats.service
	../templates/chart_viewer.service
	../templates/custom_recording.service
	../templates/extraction.service
	../templates/livestream.service
	../templates/pushed_notifications.service
	../templates/spectrogram_viewer.service
	../templates/web_terminal.service

no changes added to commit (use "git add" and/or "git commit -a")
Dropped refs/stash@{0} (7f52b577ddc2d8cc385e7728520acae6bbcfc67e)

**Comment by ehpersonal38** [comment]
Dang, you last updated over a year ago? Welcome back!

A lot has changed, including the updater script itself. So, let's try replacing your current update script with the newest one from the repo with this command:

```
wget https://raw.githubusercontent.com/mcguirepr89/BirdNET-Pi/main/scripts/update_birdnet.sh -O ~/BirdNET-Pi/scripts/update_birdnet.sh
```

Then try updating again and see if anything more productive arises.

**Reply by Rochambeaux** [reply]
/home/james/BirdNET-Pi/scripts/update_birdnet.sh: No such file or directory

**Reply by ehpersonal38** [reply]
I think you're going to have to perform a reinstall of BirdNET-Pi.

Please backup the following files if you want to keep your data:
```
~/BirdNET-Pi/birdnet.conf
~/BirdNET-Pi/scripts/birds.db
```

Then uninstall with this command:
`~/BirdNET-Pi/scripts/uninstall.sh`

Then perform a regular installation by following the instructions on this repo.
